halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/providers/AbstractProviderEnableDisableCommand.java

```diff
@@ -52,8 +53,8 @@ public String getDescription() {
 
   private void setEnable() {
     DaemonService service = Daemon.getService();
-    String currentDeployment = service.getCurrentDeployment();
-    service.setProviderEnabled(currentDeployment, getProviderName(), !noValidate, isEnable());
+    String currentDeployment = ResponseUnwrapper.get(service.getCurrentDeployment());
```

Comment:
It's another layer of indirection - the CLI portion has a class that is composed of the DaemonService.